python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -929,10 +991,18 @@ def _reorder_maps(
             [plc.types.NullOrder.AFTER, plc.types.NullOrder.AFTER],
         ).columns()
 
-    def evaluate(self, *, cache: MutableMapping[int, DataFrame]) -> DataFrame:
+    @classmethod
+    def evaluate_node(
+        cls,
+        schema: Schema,
```

Comment:
There are many `evaluate_node` implementations that do **not** need the `schema`, but **do** need all other non-child attributes. We can override `_eval_arguments` for all of these classes to strip the schema argument, but I'm still open to using a different mechanism altogether for having the `IR` subclass specify the arguments it needs to `evaluate_node`.

Reply:
Seems reasonable to me.

>`# note deliberately not splatting, it's marginally faster`

Does this prevent us from  including type-hints in the `evaluate_node` signature for that sub-class?